electron/launcher/main-process/ipc-listeners/game-client-start.js

```diff
@@ -9,16 +10,25 @@ const startGameClient = () => {
       if (error) {
         reject(error);
       }
-      resolve(stdout);
+      resolve(null);
     });
   });
 }
 
 ipc.on('start-game-client', event => {
-  
-  startGameClient().then(() => {
-    event.reply('start-game-client-success');
-  }).catch((err) => {
-    event.reply('start-game-client-fail')
-  })  
+  checkForUpdateAndDownload(event)
+    .then(() => {
+      event.reply('fm-up-to-date');
+    })
+    .then(() => {
+      return startGameClient().catch(error => {
+        throw error;
```

Comment:
do we need to catch if we just rethrow?

Reply:
Due to the nature of javascript `Promise.then()`, if we have another promise inside the `then()` and we want the error of the inner promise to make it all the way to the ending `catch()` block, we must throw it. We're just making sure the error makes it to the end and handled accordingly instead of left hanging.